### PR TITLE
feat: integrate Connectors in Self-Managed

### PR DESCRIPTION
**Pull request by tmetzke**
## What is the purpose of the change

Better integrate the Connectors in the Self-Managed documentation.
Remove Self-Managed parts from the SaaS documentation of Connectors.

## Are there related marketing activities

n/a

## When should this change go live?

Whenever ready here.

## PR Checklist

- [X] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [X] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [X] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [X] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/integration-framework/introduction.md
Comment (MaxTru) [inline docs/components/integration-framework/introduction.md:47]: The installation guide is not explicit about this IMO. I would rather link to http://localhost:3000/docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk/#runtime-logic.

File: docs/components/integration-framework/introduction.md
Comment (MaxTru) [inline docs/components/integration-framework/introduction.md:50]:
```suggestion
If you want to use the [Camunda out-of-the-box Connectors](./connectors/out-of-the-box-connectors/available-connectors-overview.md) in [Self-Managed](/self-managed/about-self-managed/), you need to set them up following the [Self-Managed Connectors Deployment guide](/self-managed/connectors-deployment/install-and-start.md).
Note that some of the out-of-the-box Connectors are licensed under the [Camunda Platform Self-Managed Free Edition license](https://camunda.com/legal/terms/cloud-terms-and-conditions/camunda-cloud-self-managed-free-edition-terms/).
```

**Review by MaxTru — CHANGES_REQUESTED** [review]

File: docs/self-managed/connectors-deployment/install-and-start.md
Comment (tmetzke) [reply docs/self-managed/connectors-deployment/install-and-start.md:13]: Well, to be consistent, we point to the different options for installing the backend which is usually either Kubernetes, Docker, or Manual installation. Currently we support Docker and Manual. There is a note on the Kubernetes section that says "Web Modeler not yet included". We could maybe add the Connectors there. It is consistent for all components then.

File: docs/self-managed/connectors-deployment/connectors-configuration.md
Comment (tmetzke) [reply docs/self-managed/connectors-deployment/connectors-configuration.md:40]: Makes sense 👍

File: docs/components/integration-framework/introduction.md
Comment (tmetzke) [reply docs/components/integration-framework/introduction.md:47]: The installation guide actually is the only place that is specific on how to consume the runtime in Self-Managed, if you ask me. Since there are different options (Docker, Manual), you have to consult those sections specifically on details then.

File: docs/self-managed/connectors-deployment/install-and-start.md
Comment (MaxTru) [reply docs/self-managed/connectors-deployment/install-and-start.md:13]: Yeah, then let's do that.

My rationale is: lets be explicit about what user can and cannot do. Otherwise they might be frustrated (_What? Only docker-compose is possible?_)

File: docs/components/integration-framework/introduction.md
Comment (tmetzke) [reply docs/components/integration-framework/introduction.md:47]: I think the best place to link to is actually http://localhost:3000/docs/next/self-managed/connectors-deployment/install-and-start. This covers backend and frontend for Connectors in Self-Managed.

File: docs/self-managed/connectors-deployment/install-and-start.md
Comment (tmetzke) [reply docs/self-managed/connectors-deployment/install-and-start.md:13]: I agree 👍

File: docs/components/integration-framework/introduction.md
Comment (tmetzke) [reply docs/components/integration-framework/introduction.md]: Fixed with [49833b0](https://github.com/camunda/camunda-platform-docs/pull/1376/commits/49833b01270769edc560abe2c9ff202f16372efb)

File: docs/components/integration-framework/introduction.md
Comment (tmetzke) [reply docs/components/integration-framework/introduction.md]: [49833b0](https://github.com/camunda/camunda-platform-docs/pull/1376/commits/49833b01270769edc560abe2c9ff202f16372efb)

File: docs/self-managed/connectors-deployment/connectors-configuration.md
Comment (tmetzke) [reply docs/self-managed/connectors-deployment/connectors-configuration.md]: Fixed with [49833b0](https://github.com/camunda/camunda-platform-docs/pull/1376/commits/49833b01270769edc560abe2c9ff202f16372efb)

File: docs/self-managed/connectors-deployment/install-and-start.md
Comment (tmetzke) [reply docs/self-managed/connectors-deployment/install-and-start.md]: Fixed with [49833b0](https://github.com/camunda/camunda-platform-docs/pull/1376/commits/49833b01270769edc560abe2c9ff202f16372efb)

File: docs/self-managed/connectors-deployment/install-and-start.md
Comment (tmetzke) [reply docs/self-managed/connectors-deployment/install-and-start.md]: Fixed with [49833b0](https://github.com/camunda/camunda-platform-docs/pull/1376/commits/49833b01270769edc560abe2c9ff202f16372efb)

File: versioned_docs/version-8.0/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md
Comment (tmetzke) [inline versioned_docs/version-8.0/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md:822]:
```suggestion
Refer to the [Self-Managed installation guide](/self-managed/connectors-deployment/install-and-start.md) for details on how to
```

File: docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md
Comment (tmetzke) [inline docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md:822]:
```suggestion
Refer to the [Self-Managed installation guide](/self-managed/connectors-deployment/install-and-start.md) for details on how to
```

File: versioned_docs/version-8.1/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md
Comment (tmetzke) [inline versioned_docs/version-8.1/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md:822]:
```suggestion
Refer to the [Self-Managed installation guide](/self-managed/connectors-deployment/install-and-start.md) for details on how to
```

**Review by tmetzke — COMMENTED** [review]
Adjust the entry link for SM installation in SDK as well.

**Review by PreetiNirwal — DISMISSED** [review]
Looks good.

**Comment by tmetzke** [comment]
@MaxTru, this needs a review from you to unblock.
@christinaausley, @akeller, I think a tech writing review makes sense here as well.

**Review by MaxTru — APPROVED** [review]
Thanks for incorporating my feedback. Looks good.

**Review by christinaausley — APPROVED** [review]
Pushed a commit with a few typo/technical adjustments. Looks good on my end 👍

**Comment by tmetzke** [comment]
@christinaausley and @akeller, some links broke after the technical review update. Can you have a look? Otherwise, it's good to go from our end. 

I enabled auto-merge. Can you have a look at where the build issue is coming from? I'm just at my phone currently, so I can't easily rebase or anything (btw, you could enable this for PRs in this repository via GitHub UI, if you haven't already done that).

Thanks a lot to everyone involved for your help here!

**Comment by christinaausley** [comment]
Sure thing @tmetzke -- I'll get this building clean.

**Pull request merged**
